packages/ui/src/components/atoms/LangDisplay.tsx

```diff
@@ -5,13 +5,26 @@ interface LangDisplayProps {
   variables?: object;
 }
 
-const getText = (templateStr: string, variables = {}) =>
+const parseVariables = (templateStr: string, variables = {}) =>
   templateStr.replace(/\${(.*?)}/g, (x, g) => (variables as any)[g] ?? '');
 
-const BaseLangDisplay: React.FC<LangDisplayProps> = ({ text, variables }) => (
-  // eslint-disable-next-line react/jsx-no-useless-fragment
-  <>{getText(text, variables)}</>
+const parseLineBreaks = (text: string) => (
+  <>
+    {' '}
+    {text.split('\n').map((item, index, arr) => (
+      <React.Fragment key={`line breaks ${item}`}>
+        {item}
+        {arr.length - 1 !== index && <br />}
+      </React.Fragment>
+    ))}
+  </>
 );
+const BaseLangDisplay: React.FC<LangDisplayProps> = ({ text, variables }) => {
+  const parsedText = parseVariables(text, variables);
+  const parsedTextWithLineBreaks = parseLineBreaks(parsedText);
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  return <>{parsedTextWithLineBreaks}</>;
+};
```

Comment:
Have you tried something like this?

```suggestion
const BaseLangDisplay: React.FC<LangDisplayProps> = ({ text, variables }) => (
  // eslint-disable-next-line react/jsx-no-useless-fragment
  <span style={{ whiteSpace: 'pre-wrap' }}>{getText(text, variables)}</span>
);
```

Reply:
This didn't work.

Reply:
This breaks the tests as we are unable to get items by roles as adding a span adds even the headings are recognized as headings. Moreover, we shouldn't add another element where we are adding a text.